src/sentry/integrations/vsts/webhooks.py

```diff
@@ -43,10 +43,13 @@ def check_webhook_secret(self, request, integration):
         assert integration.metadata['subscription']['secret'] == request.META['HTTP_SHARED_SECRET']
 
     def handle_updated_workitem(self, data, integration):
-        external_issue_key = data['resource']['workItemId']
-        assigned_to = data['resource']['fields'].get('System.AssignedTo')
-        status_change = data['resource']['fields'].get('System.State')
-        project = data['resourceContainers']['project']['id']
+        try:
+            external_issue_key = data['resource']['workItemId']
+            assigned_to = data['resource']['fields'].get('System.AssignedTo')
+            status_change = data['resource']['fields'].get('System.State')
+            project = data['resourceContainers']['project']['id']
+        except KeyError:
+            return  # In the case that there are no fields sent, no syncing can be done
```

Comment:
I feel like we should log something... not sure what

Reply:
are there any api docs that specify why this would happen? if it's unexpected/undocumented, maybe we should return a `400` response to indicate that the webhook request failed?

Reply:
You could log the JSON payload, so you can see what is inside of it?

Reply:
@macqueen I think you're being too generous to vsts's docs. Unfortunately there is not documentation there. If it's very important we can contact vsts, but their docs are woefully lacking in information, and the information that's there is typically incorrect. I will add logging to make sure we get record of when this happens. If we return 400, we will only just speed up the process that disables the webhook.

Reply:
i'm aware, that's why it was a question :)